### PR TITLE
Re-ports tg's port of vg's temperature gun

### PR DESCRIPTION
**Pull request by ToGWtF**
### Intent of your Pull Request

Before we go on, go ahead and read this fancy infographic

snazzy infographic

With this, we can nerf the xenobiology consoles (or remove entirely if you want to be that evil) without people crying too much.

#### Changelog

:cl:  

tweak: Massive tweak of the temperature gun. Now its actually useful / lethal!

/:cl:

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
this doesn't make managing slimes any easier, hosing them down with the extinguisher was literally the easiest part. The annoying part is wading through 3+ slimes to get the one you want

**Comment by ToGWtF** [comment]
It will only make managing slimes easier if we didn't have xenobio camera consoles. But they are fun for other science related things

File: code/modules/projectiles/guns/energy/temperature.dm [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+/obj/item/weapon/gun/energy/temperature/attack_self(mob/living/user as mob)
+	user.set_machine(src)
+	update_dat()
+	user << browse("<TITLE>Temperature Gun Configuration</TITLE><HR>[dat]", "window=tempgun;size=510x102")
+	onclose(user, "tempgun")
+
+/obj/item/weapon/gun/energy/temperature/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/weapon/card/emag) && !emagged)
+		emagged = 1
+		user << "<span class='caution'>You double the gun's temperature cap ! Targets hit by searing beams will burst into flames !</span>"
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline code/modules/projectiles/guns/energy/temperature.dm]: No spaces before punctuation

File: code/modules/projectiles/guns/energy/temperature.dm [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+		desc = "A gun that changes the body temperature of its targets. Its temperature cap has been hacked"
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline code/modules/projectiles/guns/energy/temperature.dm]: period

File: code/modules/projectiles/guns/energy/temperature.dm [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	var/powercost = ""
+	var/powercostcolor = ""
+	var/emagged = 0			//ups the temperature cap from 500 to 1000, targets hit by beams over 500 Kelvin will burst into flames
+	var/dat = ""
+	pin = null
+
+/obj/item/weapon/gun/energy/temperature/New()
+	..()
+	update_icon()
+	SSobj.processing |= src
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline code/modules/projectiles/guns/energy/temperature.dm]: why not START_PROCESSING(SSobj, src) instead?

File: code/modules/projectiles/guns/energy/temperature.dm [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+/obj/item/weapon/gun/energy/temperature/Destroy()
+	SSobj.processing -= src
```
Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [inline code/modules/projectiles/guns/energy/temperature.dm]: STOP_PROCESSING

File: code/modules/projectiles/guns/energy/temperature.dm (hunk 2 above)
Comment (ToGWtF) [reply code/modules/projectiles/guns/energy/temperature.dm]: I'm more concerned with it being visible emagged. Debate me on why I shouldn't remove the emagged description

Comment (1fbff5f83b23d39d38b1dfcb4cac8d9b) [reply code/modules/projectiles/guns/energy/temperature.dm]: I agree, you should remove it

**Comment by Super3222** [comment]
trick or treat!!!!1111!!

**Comment by killerx09** [comment]
I have an idea.

Why don't we just remove the Xenobio coldroom?

So this will become a thing, and now scientists can choose between fire extinguishing the slimes, shooting them with those fancy new guns or build their own cold room with NO and Freezer boards.

**Comment by Partheo** [comment]
I'm a fan. Did you make the graphic?

**Comment by ghost** [comment]
No.

**Comment by Ratvar** [comment]
Or they can RCD hole to spess in xenobio and put glass around to make coldroom that works 100% of time

File: code/modules/projectiles/guns/energy/temperature.dm [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+			if(target_temperature < temperature)
+				temperature -= (10 + 40*emagged)
+			else
+				temperature += (10 + 40*emagged)
+		else
+			temperature = target_temperature
+		update_icon()
+
+		if (istype(loc, /mob/living/carbon))
+			var /mob/living/carbon/M = loc
```
Comment (AshCorr) [inline code/modules/projectiles/guns/energy/temperature.dm:96]: Thats an interesting space there.

File: code/modules/projectiles/guns/energy/temperature.dm [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+		else			powercostcolor = "blue"
+	if(target_temperature != temperature)
+		var/difference = abs(target_temperature - temperature)
+		if(difference >= (10 + 40*emagged)) //so emagged temp guns adjust their temperature much more quickly
+			if(target_temperature < temperature)
+				temperature -= (10 + 40*emagged)
+			else
+				temperature += (10 + 40*emagged)
+		else
+			temperature = target_temperature
```
Comment (AshCorr) [inline code/modules/projectiles/guns/energy/temperature.dm:92]: Make sure you update the temperature and e_cost of the chambered round after temperature is changed. During testing the first 1-2 shots after changing the temperature would still be at the same values as the initial temperature.

File: code/modules/projectiles/guns/energy/temperature.dm [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+		else
+			src.target_temperature = max(0, src.target_temperature+amount)
+	if (istype(src.loc, /mob))
+		attack_self(src.loc)
+	src.add_fingerprint(usr)
+	return
+
+
+/obj/item/weapon/gun/energy/temperature/process()
+	switch(temperature)
```
Comment (AshCorr) [inline code/modules/projectiles/guns/energy/temperature.dm:64]: These switches should be done after temperature changes are processed.

**Review by AshCorr — CHANGES_REQUESTED** [review]

**Pull request closed without merging**